File: gherkin/dotnet/Gherkin.TokensGeneratorLib/TestTokenFormatter.cs
using System;
using System.Linq;

namespace Gherkin.TokensGenerator
{
    class TestTokenFormatter
    {
        public string FormatToken(Token token)
        {
            if (token.IsEOF)
                return "EOF";

            return string.Format("({0}:{1}){2}:{3}/{4}/{5}", 
                token.Location.Line,
                token.Location.Column,
                token.MatchedType,
                token.MatchedKeyword,
                token.MatchedText,
                token.MatchedItems == null ? "" : string.Join(",", token.MatchedItems.Select(i => i.Column + ":" + i.Text))
                );
        }
    }
}

File: gherkin/dart/lib/src/tokens/TestTokenFormatter.dart
import 'package:gherkin/language.dart';

class TestTokenFormatter
{
  String formatToken(Token token)
  {
    if (token.isEof) {
      return 'EOF';
    }
    var items = token.matchedItems.map((element) => '${element.column}:${element.text}');
    var matchedItems = items.isEmpty ? '' : items.join(',');
    var matchedType = token.matchedType.toString().split('.').last;
    var matchedKeyword = token.matchedKeyword.isEmpty ? '' : token.matchedKeyword;
    var matchedText = token.matchedText.isEmpty ? '' : token.matchedText;
    return '(${token.location.line}:${token.location.column})'
        '$matchedType:$matchedKeyword/$matchedText/$matchedItems';
  }
}